src/payment/strategies/zip/zip-payment-strategy.ts

```diff
@@ -0,0 +1,159 @@
+            this._store.dispatch(this._paymentMethodActionCreator.loadPaymentMethod(this._methodId))
+                .then(state => {
+                    if (!this._zipClient) {
+                        throw new NotInitializedError(NotInitializedErrorType.PaymentNotInitialized);
+                    }
+
+                    this._zipClient.Checkout.init(this._getZipPayload());
+                });
+
+            // Lightbox closing event listener
+            return new Promise((resolve, reject) => {
+                this._lightboxEvents$
+                    .pipe(take(1))
+                    .subscribe((event: { type: ZipModalEvent }) => {
+                        if (event.type === ZipModalEvent.CancelCheckout) {
+                            reject(new PaymentMethodCancelledError());
+                        }
+                        if (event.type === ZipModalEvent.CheckoutApproved) {
+                            resolve(this._submitOrder(payload, options));
+                        }
+                    });
+            });
+        }, { methodId: this._methodId }), { queueId: 'widgetInteraction' });
+    }
+
+    private _getZipPayload(): ZipPayload {
+        return {
+            onComplete: this._handleResponse,
+            onCheckout: this._fetchPayload,
+        };
+    }
+
+    private _submitOrder(payload: OrderRequestBody, options?: PaymentRequestOptions): Promise<InternalCheckoutSelectors> {
+        const order = { useStoreCredit: payload.useStoreCredit };
+
+        if (!this._zipLightboxResponse) {
+            throw new MissingDataError(MissingDataErrorType.MissingPaymentMethod);
+        }
+
+        const paymentPayload = {
+            methodId: this._methodId,
+            paymentData: { nonce: this._zipLightboxResponse.checkoutId } as NonceInstrument,
+        };
+
+        return this._store.dispatch(this._orderActionCreator.submitOrder(order, options))
+            .then(() => this._store.dispatch(this._paymentActionCreator.submitPayment(paymentPayload)));
+    }
+
+    @bind
+    private _fetchPayload(resolve: (response: ZipPostResponse) => void): void {
+        this._store.dispatch(this._paymentMethodActionCreator.loadPaymentMethod(this._methodId))
+            .then(state => {
+                const paymentMethod = state.paymentMethods.getPaymentMethod(this._methodId);
+
+                if (!paymentMethod || !paymentMethod.clientToken) {
+                    throw new MissingDataError(MissingDataErrorType.MissingPaymentMethod);
+                }
+
+                resolve(JSON.parse(paymentMethod.clientToken));
+            });
+    }
+
+    @bind
+    private _handleResponse(response: ZipResponse): void {
+        if (response.state === ZipModalEvent.CancelCheckout) {
+            this._lightboxEvents$.next({ type: ZipModalEvent.CancelCheckout });
+
+            return;
+        } else if (response.state === ZipModalEvent.CheckoutApproved) {
```

Comment:
If there's a `return` statement above, you don't actually need to write `else` because if the condition is true above, the code below won't get executed anyway.